Issue #1: compatibility with 5250, 5200XXX, N2520, N8800Prov2

dra opened the pull request:
Hi.

These commits adjust the merge of issue#1 to be compatible with our 3 different models: N2520, N5200XXX, N8800Prov2

mhoogveld committed:
Issue #1: Add support for NASBOX5G2
Merged the adjustments made by Chris into the check script. (Added memory check, changed certain json calls and response parsing)
Added method to check multiple uri's for information. First one to generate a valid response is used.

mhoogveld committed:
Completed merging N5500 check code with NASBOX5G2 check code.
Added debug option which displays all json responses to assist in supporting other Thecus(-based) devices.

mhoogveld committed:
Changed CURLINFO_RESPONSE_CODE to CURLINFO_HTTP_CODE.
CURLINFO_RESPONSE_CODE is supported only by PHP 5.5 and above, while CURLINFO_HTTP_CODE has been supported in earlier versions.

dra committed:
Accept 'none' as a return for fan check on some models.

dra committed:
Some models return a different disk array structure.

dra committed:
Improve disk status handling.

dra committed:
Correct order of diskno and trayno arguments.

dra committed:
Add another URI to smart checks, 'diskno' vs. 'disk_no' ...

dra committed:
Merge.

dra committed:
Remember successful smart check URI to reduce the number of requests on large storages.

mhoogveld commented:
Thanks for the check and code-update!
I have added some very small improvements and cleaned up the code a bit.

I'm not so sure that lines 1087 +18 are correct.
You are reusing the $data variable of the outer for loop in the inner for loop in checkDiskHealth().
The comment in the commit was "Some models return a different disk array structure." so I'm not sure what you were trying to accomplish. 
Maybe you can check this bit again?

The piece of code in question:
<pre>
// Fill the diskList array with all disks found in the one or two dimensional diskInfo array
$diskList = array();
foreach ($diskInfo->disk_data as $disk) {
    if (isset($disk->disks)) {
        foreach ($disk->disks as $d) {
            $diskList[$d->tray_no][$d->disk_no] = $d;
        }
    } else if (!isset($diskList[$disk->trayno])) {
        foreach ($diskInfo->disk_data as $disk) {
            if (!isset($diskList[$disk->trayno])) {
                $diskList[$disk->trayno] = array();
            }
            $diskList[$disk->trayno][$disk->diskno] = $disk;
        }
    } else {
        $diskList[$disk->trayno][$disk->diskno] = $disk;
    }
}
</pre>

dra commented:
Hi.

Sorry, was obviously too late in the night when I coded that...
Created the PR#4 which simplifies the code a lot.

Bye, Daniel


On 04/30/2018 02:50 PM, mhoogveld wrote:
> Thanks for the check and code-update!
> I have added some very small improvements and cleaned up the code a bit.
> 
> I'm not so sure that lines 1087 +18 are correct.
> You are reusing the $data variable of the outer for loop in the inner
> for loop in checkDiskHealth().
> The comment in the commit was "Some models return a different disk array
> structure." so I'm not sure what you were trying to accomplish.
> Maybe you can check this bit again?
> 
> The piece of code in question:
> 
> // Fill the diskList array with all disks found in the one or two dimensional diskInfo array
> $diskList = array();
> foreach ($diskInfo->disk_data as $disk) {
>     if (isset($disk->disks)) {
>         foreach ($disk->disks as $d) {
>             $diskList[$d->tray_no][$d->disk_no] = $d;
>         }
>     } else if (!isset($diskList[$disk->trayno])) {
>         foreach ($diskInfo->disk_data as $disk) {
>             if (!isset($diskList[$disk->trayno])) {
>                 $diskList[$disk->trayno] = array();
>             }
>             $diskList[$disk->trayno][$disk->diskno] = $disk;
>         }
>     } else {
>         $diskList[$disk->trayno][$disk->diskno] = $disk;
>     }
> }
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/mhoogveld/check_thecus_nas/pull/3#issuecomment-385389155>,
> or mute the thread
> <https://github.com/notifications/unsubscribe-auth/AAH891GTZC75Si0fnN-2-18Yp0IlaNKpks5ttwizgaJpZM4TsAkf>.
> 


-- 
bytemine GmbH - Im Technologiepark 4 - 26129 Oldenburg
Support: 0441-30 91 97-42 - Vertrieb: 0441-30 91 97-69
Geschäftsführung: Daniel Rauer - Dr. Christoph Niemann
Amtsgericht in Oldenburg --- Handelsregister B 203 940
https://www.bytemine.net --- Steuernummer: DE267286778
